### DISCUSSION TITLE
<cac:ContractingPartyType> in notice SubType 14

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#407] [Q&A] [answered]
Hi Labs
in notice SubtYPE 14 cac:ContractingPartyType gets in this way : 
<cac:ContractingPartyType>
			<cbc:PartyTypeCode listName="buyer-legal-type">body-pl</cbc:PartyTypeCode>
			<cbc:PartyTypeCode listName="buyer-contracting-type">cont-ent</cbc:PartyTypeCode>
		</cac:ContractingPartyType>
because the tag cac:ContractingPartyType the second time is found : 
foundElementsOpt = Optional.ofNullable(XmlUtils.evaluateXpathAsNodeList(xpathInst, previousElem, xpathExprOpt, fieldId));
of course this gets an xsd error 
I guess, from the example notice, that it should be in this way : 
<cac:ContractingPartyType>
			<cbc:PartyTypeCode listName="buyer-legal-type">la</cbc:PartyTypeCode>
		</cac:ContractingPartyType>
		<cac:ContractingPartyType>
			<cbc:PartyTypeCode listName="buyer-contracting-type">not-cont-ent</cbc:PartyTypeCode>
		</cac:ContractingPartyType>
		<cac:ContractingActivity>
			<cbc:ActivityTypeCode listName="authority-activity">gen-pub</cbc:ActivityTypeCode>
		</cac:ContractingActivity>

I there a sort of "rule" for which the tag must be created again ? 
Thanks and Regards
Renato

**Comment by rouschr** [comment, marked as answer]
You are guessing correctly.
We have seen this problem in another discussion.
Here the problem is with the code trying to reuse existing XML elements which is OK in other cases,
For the moment the logic would be to say that if the field is not repeatable (in fields.json): recreate the entire absolute xpath element chain.
```
    "xpathRelative" : "cac:ContractingPartyType/cbc:PartyTypeCode[@listName='buyer-contracting-type']",
```
If you cannot find it using the xpath: 
```
"cac:ContractingPartyType/cbc:PartyTypeCode[@listName='buyer-contracting-type']"
```
Then create a new "cac:ContractingPartyType" and "cbc:PartyTypeCode" inside of it.
